builder/builder.go

```diff
@@ -51,223 +71,198 @@ func (b *Builder) SetMirrorList() error {
+}
+
 // Init initializes the container
 func (b *Builder) Init() error {
 	// No need to initialize an initialized container
-
-	exists := backend.CheckContainerExists(b.Path)
-	if exists {
+	if backend.CheckContainerExists(b.Path) {
 		b.ContainerPath = path.Join(b.Path, "root")
 		b.Container.SetPath(b.ContainerPath)
 		return nil
 	}
-
+	//
 	ContainerPath, err := b.Backend.Setup()
 	if err != nil {
 		return err
 	}
-
 	// Point the container to the correct container
 	b.ContainerPath = ContainerPath
 	b.Container.SetPath(b.ContainerPath)
-
-	err = b.Bootstrap.Init(b.ContainerPath)
-	if err != nil {
-		return err
-	}
-
-	err = b.SetupConfig()
-	if err != nil {
+	//
+	if err := b.Bootstrap.Init(b.ContainerPath); err != nil {
 		return err
 	}
-
-	err = utils.CreateFile(b.ContainerPath, ".arch-chroot", []byte("v4"), 0644)
-	if err != nil {
+	//
+	if err := b.SetupConfig(); err != nil {
 		return err
 	}
-	err = utils.CreateFile(b.ContainerPath, "/etc/locale.conf", []byte("LANG=en_US.UTF-8"), 0644)
-	if err != nil {
+	// Create the initial files in the container
+	if err := b.CreateFiles(initFileMap); err != nil {
 		return err
 	}
-	err = utils.CreateFile(b.ContainerPath, "/etc/locale.gen", []byte("en_US.UTF-8 UTF-8"), 0644)
-	if err != nil {
+	// Generate locale files in the container
+	if err := b.Container.Exec("locale-gen"); err != nil {
 		return err
 	}
-	err = b.Container.Exec("locale-gen")
-	if err != nil {
+	//
+	if err := utils.SetupCacheDirs(b.Container, b.PacmanConf); err != nil {
 		return err
 	}
-
-	err = utils.SetupCacheDirs(b.Container, b.PacmanConf)
-	if err != nil {
-		return err
-	}
-
-	err = b.Container.Exec("pacman -Syu --noconfirm base-devel")
-	if err != nil {
+	// Upgrade packages in the container
+	if err := b.Container.Exec("pacman -Syu --noconfirm base-devel"); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (b *Builder) SetupChrootConfig() error {
+// ContainerFunction can perform actions on a Container
+type ContainerFunction func(container.Container) error
 
-	err := b.SetupGnupg()
-	if err != nil {
-		return err
-	}
+// ContainerFunctions is a slice of ContainerFunction
+type ContainerFunctions []ContainerFunction
 
-	err = utils.SetupUser(b.Container)
-	if err != nil {
-		return err
-	}
-
-	err = utils.SetupMakepkgDirectories(b.Container)
-	if err != nil {
-		return err
+// WithContainer takes a slice of functions that takes a Container
+// as an argument and applies them to the container within the Builder.
+// If there is an error, the rest of the functions will not be called.
+func (b *Builder) WithContainer(cfs ContainerFunctions) error {
+	c := b.Container
+	for _, cf := range cfs {
+		if err := cf(c); err != nil {
+			return err
+		}
 	}
+	return nil
+}
 
-	err = utils.SetupMakepkg(b.Container)
-	if err != nil {
+// SetupChrootConfig -
+func (b *Builder) SetupChrootConfig() error {
+	if err := b.SetupGnupg(); err != nil {
 		return err
 	}
-	return nil
+	return b.WithContainer(ContainerFunctions{
```

Comment:
`ContainerActions` may be a better name

Reply:
Yes, thats a better name. I wonder if some of these functions can be refactored out of the builder struct completely into it's own file.